### PR TITLE
Read objects in ERA files

### PR DESCRIPTION
**Pull request by rolfyone**
Added some chain validation, can now pass multiple files and they'll be read in order.

This shows enough of a proof, and core concepts.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java [hunk 1]
```diff
@@ -90,18 +105,132 @@ private void getStateIndices() {
...
+  }
+
+  private void verifyBlocksWithReferenceState(
+      BeaconState verifiedState, final SignedBeaconBlock previousArchiveLastBlock)
+      throws IOException {
+    currentSlot = blockIndices.getStartSlot();
+    Bytes32 lastRoot = Bytes32.ZERO;
+    int populatedSlots = 0;
+    int emptySlots = 0;
+    for (int i = 0; i < blockIndices.getCount(); i++) {
```
Comment (tbenr) [inline data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java:175]: maybe moving all to `long` would make CodeQL happy

File: data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java [hunk 2]
```diff
@@ -111,10 +240,25 @@ final void close() throws IOException {
...
 
   public static void main(String[] args) throws IOException {
-    final Path p = Path.of(args[0]);
-    final EraFile f = new EraFile(p);
-    f.readEraFile();
-    f.printStats();
-    f.close();
+    Spec spec = null;
+    SignedBeaconBlock lastBlock = null;
+    for (String filename : args) {
```
Comment (tbenr) [inline data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java]: `final`

File: data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java [hunk 3]
```diff
@@ -111,10 +240,25 @@ final void close() throws IOException {
   }
 
   public static void main(String[] args) throws IOException {
```
Comment (tbenr) [inline data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java]: `final`

File: data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java [hunk 4]
```diff
@@ -111,10 +240,25 @@ final void close() throws IOException {
...
+    SignedBeaconBlock lastBlock = null;
+    for (String filename : args) {
+      final Path p = Path.of(filename);
+      if (spec == null) {
+        final String network =
+            Arrays.stream(p.getFileName().toString().toLowerCase(Locale.ROOT).split("-", 2))
+                .findFirst()
+                .orElseThrow();
+        System.out.println("Loading network: " + network);
+        spec = SpecFactory.create(network);
```
Comment (tbenr) [inline data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java]: we could have this in a private method like `getNetworkFromEraFilename` and we can call it once outside the loop using the first file from the arg

**Review by tbenr — APPROVED** [review]
LGTM. Just some nits.

File: data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java (hunk 1 above)
Comment (rolfyone) [reply data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java:175]: or fix object :) all good

File: data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java (hunk 4 above)
Comment (rolfyone) [reply data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java]: Fair, I guess I was more thinking this is just a test file but I'll do that and add a usage if there's no args.

Comment (tbenr) [reply data/dataexchange/src/main/java/tech/pegasys/teku/data/eraFileFormat/EraFile.java]: Ah yes...:)

**Pull request merged**